docusaurus/docs/how-to-guides/app-plugins/add-backend-component.md

```diff
@@ -0,0 +1,145 @@
+
+# Use cases for backend components in app plugins
+
+- Use custom authentication methods and/or authorization checks that aren't supported in Grafana.
+- Running workloads in the background
+- Connect to non-HTTP services that normally can't be connected to from a browser
+
+# Add a backend component to an app plugin
+
+## Prerequisites
+
+- Go ([Version](https://github.com/grafana/plugin-tools/blob/main/packages/create-plugin/templates/backend/go.mod#L3))
+- [Mage](https://magefile.org/)
+- [LTS](https://nodejs.dev/en/about/releases/) version of Node.js
+- [Docker](https://docs.docker.com/get-docker/)
+
+## Create a new app plugin
+
+<CreatePlugin pluginType="app" />
+
+## Anatomy of a backend plugin
+
+<BackendPluginAnatomy pluginType="app" />
+
+## Troubleshooting
+
+### Grafana doesn't load my plugin
+
+Ensure that Grafana has been started in development mode. If you are running Grafana from source, you'll need to add the following line to your `conf/custom.ini` file (if you don't have one already, go ahead and create this file before proceeding):
+
+```ini
+app_mode = development
+```
+
+You can then start Grafana in development mode by running `make run & make run-frontend` in the Grafana repository root.
+
+If you are running Grafana from a binary or inside a Docker container, you can start it in development mode by setting the environment variable `GF_DEFAULT_APP_MODE` to `development`.
+
+By default, Grafana requires backend plugins to be signed. To load unsigned backend plugins, you need to
+configure Grafana to [allow unsigned plugins](https://grafana.com/docs/grafana/latest/administration/plugin-management/#allow-unsigned-plugins).
+For more information, refer to [https://www.action.com/nl-nl/p/1325690/c-c-autowax-en-polijstmiddel/Plugin signature verification](https://grafana.com/docs/grafana/latest/administration/plugin-management/#backend-plugins).
+
+## Access App settings
+
+Settings are part of the `AppInstanceSettings` struct. They are passed to the app plugin constructor as the second argument.
+
+```go title="src/app.go"
+func NewApp(ctx context.Context, settings backend.AppInstanceSettings) (instancemgmt.Instance, error) {
+  jsonData := settings.JSONData // json.RawMessage
+  secureJsonData := settings.DecryptedSecureJSONData // map[string]string
+}
+```
+
+You can also get the settings from a request `Context`
+
+```go title="src/resources.go"
+func (a *App) handleMyRequest(w http.ResponseWriter, req *http.Request) {
+  pluginConfig := backend.PluginConfigFromContext(req.Context())
+  jsonData := pluginConfig.AppInstanceSettings.JSONData // json.RawMessage
+}
+```
+
+## Add a custom endpoint to your app plugin
+
+### ServeMux (recommended)
+
+Your scaffoled app plugin already has a default CallResource that uses [ServeMux](https://pkg.go.dev/net/http#ServeMux).
+
+You can add custom endpoints to your app plugin in the `resources.go` file. E.g.:
```

Comment:
You probably also need to mention that the App needs to implement the CallResourceHandler interface:

```go
type App struct {
	backend.CallResourceHandler
}

// NewApp creates a new example *App instance.
func NewApp(_ context.Context, _ backend.AppInstanceSettings) (instancemgmt.Instance, error) {
	var app App

	// Use a httpadapter (provided by the SDK) for resource calls. This allows us
	// to use a *http.ServeMux for resource calls, so we can map multiple routes
	// to CallResource without having to implement extra logic.
	mux := http.NewServeMux()
	app.registerRoutes(mux)
	app.CallResourceHandler = httpadapter.New(mux)

	return &app, nil
}
```

Reply:
Good point! I omitted it because it is part of the scaffolded code but no harm in adding it to the docs.